src/PairedRouting.php

```diff
@@ -772,6 +772,38 @@ public function add_permalink_settings_notice() {
 		<?php
 	}
 
+	/**
+	 * Update canonical URL for legacy reader paired urls.
+	 *
+	 * @param string $redirect_url Canonical redirect URL.
+	 *
+	 * @return string Updated canonical URL.
+	 */
+	public function maybe_update_paired_url( $redirect_url ) {
+
+		global $wp_rewrite;
+
+		/**
+		 * To address issue AMP version of a comment page.
+		 * Reference: https://github.com/ampproject/amp-wp/issues/6355
+		 */
+		if ( ! empty( $redirect_url ) &&
```

Comment:
This could also check if the `cpage` query var is present since this is only an issue for such requests:

```suggestion
		if ( ! empty( $redirect_url ) &&
			get_query_var( 'cpage' ) &&
```